org.eclipse.xtext.util/src/org/eclipse/xtext/util/PolymorphicDispatcher.java

```diff
@@ -353,7 +353,23 @@ protected RT handleAmbigousMethods(List<MethodDesc> result, Object... params) {
 		Collections.sort(cachedDescriptors, new Comparator<MethodDesc>() {
 			@Override
 			public int compare(MethodDesc o1, MethodDesc o2) {
-				return PolymorphicDispatcher.this.compare(o1, o2);
+				int compare = PolymorphicDispatcher.this.compare(o1, o2);
+				if (compare != 0) {
+					return compare;
+				}
+				Class<?>[] p1 = o1.getParameterTypes();
+				Class<?>[] p2 = o2.getParameterTypes();
+				int to = Math.min(p1.length, p2.length);
+
+				for (int i = 0; i < to; i++) {
+					final String n1 = p1[i].getName();
+					final String n2 = p2[i].getName();
+					compare = n1.compareTo(n2);
+					if (compare != 0) {
+						return compare;
+					}
+				}
+				return compare;
```

Comment:
It looks fine.
But I was wondering whether this is better/more efficient than simply comparing the method desc string representation.
Just a curiosity